### ISSUE TITLE
UndefinedVariable first assigned in for loop, used in loop expression

### ISSUE THREAD

**Issue by kevinoid** [sirbrillig/phpcs-variable-analysis#261]
With PHP_CodeSniffer VariableAnalysis v2.11.3 and  PHP_CodeSniffer 3.7.1, if a variable which is first assigned in the body of a for loop is used in the third expression of the for loop (i.e. the "increment expression" of the for loop) it causes a VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable warning.  To reproduce:

1. Create `fwrite_stream.php` with the `fwrite_stream` function from [`fwrite` Notes](https://www.php.net/manual/en/function.fwrite.php#refsect1-function.fwrite-notes):
   ```php
   <?php
   function fwrite_stream($fp, $string) {
       for ($written = 0; $written < strlen($string); $written += $fwrite) {
           $fwrite = fwrite($fp, substr($string, $written));
           if ($fwrite === false) {
               return $written;
           }
       }
       return $written;
   }
   ?>
   ```
2. Install and configure PHP_CodeSniffer VariableAnalysis:
   ```sh
   composer init --no-interaction --name kevinoid/fwrite_stream
   composer config allow-plugins.dealerdirect/phpcodesniffer-composer-installer true
   composer require --dev dealerdirect/phpcodesniffer-composer-installer
   composer require --dev sirbrillig/phpcs-variable-analysis
   echo '<ruleset><rule ref="VariableAnalysis"/></ruleset>' >.phpcs.xml
   ```
3. Run PHP_CodeSniffer:
   ```sh
   composer exec -- phpcs -s --report=emacs fwrite_stream.php
   ```
   and observe the output:

       /path/to/fwrite_stream.php:3:64: warning - Variable $fwrite is undefined. (VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable)

This appears to be a false-positive, since although the third expression of the for loop appears before the body in the source, it is always evaluated after the body which defines the variable.

Thanks for considering, and for creating/maintaining this project!
Kevin

**Comment by sirbrillig** [comment]
You're right! It seems obvious now but I would not have thought of that before. I'll see if I can modify the code to detect that pattern and allow it.
